diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,9 @@ if(POLICY CMP0091)
 endif()
 
 project(ninecraft LANGUAGES C CXX)
+set(GLAD_DIR $ENV{glad})
+set(STB_DIR $ENV{stb})
+
 
 if(NOT CMAKE_SIZEOF_VOID_P EQUAL 4)
     message(FATAL_ERROR
@@ -57,7 +60,7 @@ endif()
 
 include(CTest)
 
-include_directories(${PROJECT_SOURCE_DIR}/stb)
+include_directories(${STB_DIR})
 
 if(WIN32)
     # The Windows distribution is a single executable.  Libraries without an
@@ -117,11 +120,11 @@ SET(ZLIB_BUILD_SHARED OFF CACHE BOOL "Disable the unused shared zlib library" FOR
 SET(ZLIB_BUILD_STATIC ON CACHE BOOL "Build the static zlib library" FORCE)
 SET(ZLIB_INSTALL OFF CACHE BOOL "Disable zlib installation targets" FORCE)
 
-add_subdirectory(glad/cmake)
+add_subdirectory(${GLAD_DIR}/cmake ${GLAD_DIR})
 glad_add_library(glad REPRODUCIBLE LOADER API egl=1.5 gl:compatibility=2.0)
-add_subdirectory(zlib)
+find_package(ZLIB REQUIRED)
 add_subdirectory(ancmp)
-add_subdirectory(SDL)
+find_package(SDL2 REQUIRED)
 add_subdirectory(ninecraft)
 
 if(MSVC)
diff --git a/ninecraft/CMakeLists.txt b/ninecraft/CMakeLists.txt
--- a/ninecraft/CMakeLists.txt
+++ b/ninecraft/CMakeLists.txt
@@ -4,9 +4,9 @@ project(ninecraft-core LANGUAGES C)
 
 file(GLOB_RECURSE SOURCE_FILES CONFIGURE_DEPENDS src/*.c)
 
-add_executable(ninecraft ${SOURCE_FILES} ../stb/stb_vorbis.c)
+add_executable(ninecraft ${SOURCE_FILES} ${STB_DIR}/stb_vorbis.c)
 target_include_directories(ninecraft PRIVATE include/)
-target_link_libraries(ninecraft PRIVATE ${CMAKE_DL_LIBS} zlibstatic SDL2::SDL2 SDL2main glad ancmp)
+target_link_libraries(ninecraft PRIVATE ${CMAKE_DL_LIBS} ZLIB::ZLIB SDL2::SDL2 SDL2main glad ancmp)
 
 if(WIN32)
     target_link_libraries(ninecraft PRIVATE iphlpapi)
